OloEngine/src/OloEngine/Renderer/Commands/FrameDataBuffer.cpp

```
@@ -147,6 +162,106 @@ namespace OloEngine
+            }
+            return INVALID_RENDER_STATE_INDEX;
+        }
+
+        u16 index = m_RenderStateCount;
+        m_RenderStates[index] = state;
+        ++m_RenderStateCount;
+
+        return index;
+    }
+
+    const PODRenderState& FrameDataBuffer::GetRenderState(u16 index) const
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)